cores/esp32/esp32-hal-ledc.c

```diff
@@ -58,15 +60,25 @@ static bool ledcDetachBus(void * bus){
 
 bool ledcAttachChannel(uint8_t pin, uint32_t freq, uint8_t resolution, uint8_t channel)
 {
-    if (channel >= LEDC_CHANNELS || resolution > LEDC_MAX_BIT_WIDTH)
-    {
-        log_e("Channel %u is not available! (maximum %u) or bit width too big (maximum %u)", channel, LEDC_CHANNELS, LEDC_MAX_BIT_WIDTH);
+    if(channel >= LEDC_CHANNELS || ledc_handle.used_channels & (1UL << channel)){
+        log_e("Channel %u is not available (maximum %u) or already used!", channel, LEDC_CHANNELS);
+        return false;
+    }
+    
+    if (resolution > LEDC_MAX_BIT_WIDTH){
+        log_e("LEDC resolution too big (maximum %u)", LEDC_MAX_BIT_WIDTH);
```

Comment:
```suggestion
        log_e("LEDC resolution is zero or it is too big (maximum %u)", LEDC_MAX_BIT_WIDTH);
```
Message about zero resolution.